src/components/sidebar/routePathView/RoutePathView.tsx

```diff
@@ -107,23 +156,41 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
         );
     }
 
-    async componentDidMount() {
-        if (this.props.isAddingNew) {
-            await this.initializeAsAddingNew();
-        } else {
-            await this.fetchRoutePath();
+    private isFormValid = () => {
+        return !Object.values(this.state.invalidFieldsMap)
+            .some(fieldIsValid => !fieldIsValid);
```

Comment:
Maybe change this to:
return Object.values(this.state.invalidFieldsMap).every(fieldIsValid => fieldIsValid).

Reply:
Hmm, this is actually faster this way.